src/js/components/Button/Button.js

```diff
@@ -83,9 +84,16 @@ const getIconColor = (paths = [], theme, colorProp, kind) => {
 
       let color;
       if (obj?.icon?.props?.color) color = obj.icon.props.color;
+      // if no icon defined for this state, see if there is an icon
+      // with color defined at one higher level
+      else if (paths[index + 1]) {
+        const parts = paths[index + 1].split('.');
+        while (baseObj && parts.length) obj = baseObj[parts.shift()];
+        if (obj?.icon?.props?.color) color = obj.icon.props.color;
+      }
       // use passed in color for text if the theme doesn't have
       // background or border color
-      else
+      if (!color)
```

Comment:
Now we are getting an unexpected change in font color, presumably because the background color is now calculated differently?

![Screen Shot 2022-08-25 at 3 57 34 PM](https://user-images.githubusercontent.com/1756948/186776600-d9e7f3fa-96d7-45d3-9d26-f26f38444f99.png)

Reply:
This issue is already existing on master. I have file a separate issue for it: https://github.com/grommet/grommet/issues/6301

Reply:
👍